app/code/Magento/Deploy/Package/PackagePool.php

```diff
@@ -241,6 +270,47 @@ private function ensureRequiredLocales(array $options)
         }
     }
 
+    /**
+     * Get used store and admin user locales.
+     *
+     * @return array
+     * @throws InvalidArgumentException if unknown locale is provided by the store configuration
+     */
+    private function getUsedLocales()
+    {
+        $usedLocales = array_merge(
+            $this->storeView->retrieveLocales(),
+            $this->getAdminUserInterfaceLocales()
```

Comment:
Let's assume we have only en_us locale on backend, but 5 locales on the frontend. Seems like this change will cause generating static content for those 5 locales that are used on frontend, which is redundant.
I think we need to differentiate frontend and backend locales.

Please correct me if I'm wrong.

Reply:
This is indeed a welcome improvement and I will look into adding it to the PR. If it can be done without too big of an impact.

Reply:
This improvement is now part of the PR